modmail/dispatcher.py

```diff
@@ -0,0 +1,167 @@
+import asyncio
+import bisect
+import logging
+from typing import Callable, Coroutine, Dict, List, Optional, Union
+
+from modmail import ModmailLogger
+
+logger: ModmailLogger = logging.getLogger(__name__)
+
+CoroutineFunction = Callable[..., Coroutine]
+
+
+class Dispatcher:
```

Comment:
Is there a way to close a dispatcher, to remove all events from it safely and ensure nothing was being awaited or uncollected?

Reply:
That is something I should add.

Reply:
No, but that's because it's just a flow-through. Handling safe closure of all dispatch calls should ensure the dispatcher is cleaned up afterwards.